### ISSUE TITLE
Can't Rename Scenes After Creating Them

### ISSUE THREAD

**Issue by DAOQ-U** [obsproject/obs-studio#3044]
When I click the '+' button on the bottom of the Scenes box and Type a Name for the Scene, then add my sources to the scene, then later decide I want to use a different name for the scene, I highlight the scene name and right click and select Rename from the popup menu.  The text does become editable and I type the new name and press the return key.  At that point the name goes back to the original name and is not renamed as it should have been.  The only thing I can do to get around this is to recreate the entire scene using the correct name in the first place and then delete the old one I no longer need.

I only have 1 Scene Collection
Using MacOS Catalina 10.15.5 - All Updates
Mackbook Pro (2017 - 15Inch - 3.1 Ghz Quad Core i7 - 16GB Ram)
OBS Studio 25.0.8 (64-Bit) - Downloaded Main Installer
VLC 3.0.10 - 64 Bit
Latest Version of Final Cut Pro

**Comment by dodgepong** [comment]
This is a known bug where pressing the Enter key after renaming a scene reverts the name. If you click somewhere else in OBS to deselect the name instead of pressing Enter, the new name will stick.

**Comment by DAOQ-U** [comment]
Ok, thanks.  Didn’t know this was a known bug.

**Comment by dodgepong** [comment]
We can keep this open, I don't think there's an issue open for this already.

In the future, though, please be sure to completely fill out the issue template.

**Comment by unknowndomain** [comment]
The solution is to press Command + Enter rather than just Enter.

I assume that pressing enter starts a new renaming 'session' each time, this doesn't happen on Windows because you use F2 to rename then enter to accept.

**Comment by mtippett** [comment]
The root cause of this bug appears to be in window-basic-main.cpp#L315 where the UI element has “Qt::Key_Return” added as a shortcut. This appears to cause the Edit to be called again when the user hits “Enter” once more. This explanation is incomplete as it doesn’t provide for a reason for the slightly different behaviors between the Source and Scene.

Commenting out the lines allows it to work as expected. Investigation details are on the livedebug.dev page for [this issue](https://www.livedebug.dev/obs-studio-3044/README.html).  Have a pending discussion on #development on discord to understand the next steps to resolve.  

Commenting out the lines above remove the <Return> to edit functionality, but when editing through the context menu allows the edit to be saved with "enter".

**Comment by mtippett** [comment]
I'm creating a PR request to resolve this.  I couldn't find any references in the [macOS User Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/macos/overview/themes/) covering the default edit behaviors.  Most of the Apple interfaces do not allow editing by hitting enter.  You either select the item and hit the gear icon or right click to rename.  

Confirmed with @admshao that there is no particular attraction to that behavior.

**Comment by unknowndomain** [comment]
As a macOS use I strongly object to removing this feature entirely as a solution to fixing a bug. 

I'm not sure that the HIG should always be the golden reference as to the only way to do things. 

Ever place you can rename something in macOS Finder being the primary example, but there are others. You can navigate up and down to select a file then press enter to edit and enter to save.

**Comment by Scrxtchy** [comment]
I think I've traced the issue to this code block
https://github.com/obsproject/obs-studio/blob/master/UI/source-tree.cpp#L398-L417

I'm not a mac user, but I can assume the difference between enter and return functionality.
As one condition checks for a LineEditChanged, as saves the data, the other checks for a LineEditCancel, and destorys the edit.
@jp9000 added this back in d5d8492b

**Comment by RytoEX** [comment]
> I think I've traced the issue to this code block
> https://github.com/obsproject/obs-studio/blob/master/UI/source-tree.cpp#L398-L417
> 
> I'm not a mac user, but I can assume the difference between enter and return functionality.
> As one condition checks for a LineEditChanged, as saves the data, the other checks for a LineEditCancel, and destorys the edit.
> @jp9000 modified this back in [d5d8492](https://github.com/obsproject/obs-studio/commit/d5d8492bb327753ed48d363481e3012e2c358825), where it used to explicitly use Key values

@Scrxtchy 
That seems unlikely since that code is for the Source list, not the Scene list.  This issue is about renaming _scenes_, not sources.  Renaming sources works just fine.  However, when renaming a scene, if you press the rename shortcut key (Return/Enter on macOS, F2 on Windows/Linux), the rename action will restart, discarding any changes you've made to the scene's name.

I've discussed this at length in the Discord's development channel with @mtippett.  The Source list and the Scene list are handled by different classes.  The Source list has a lot of extra code handling edit start, stop, and finished events to prevent this sort of thing occurring there.  The Scene list does not have the same amount of code for it.  As implied by the commit you linked, and as I discussed on Discord, this comes down to the fact that editing is handled by Qt item delegates (editors), and any code preventing the restart of the edit action would have to be linked to the item delegate of the SceneTree, which is a subclassed QListWidget.
